x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/select_rule_type/index.tsx

```diff
@@ -48,7 +48,7 @@ export const SelectRuleType: React.FC<SelectRuleTypeProps> = memo(
     const setNewTerms = useCallback(() => setType('new_terms'), [setType]);
     const setEsql = useCallback(() => setType('esql'), [setType]);
 
-    const isEsqlRuleTypeEnabled = useIsEsqlRuleTypeEnabled();
+    const { isEsqlRuleTypeEnabled } = useEsqlAvailability();
```

Comment:
Rule type icon should be displayed when editing rule

<img width="1234" alt="Screenshot 2024-04-30 at 17 37 29" src="https://github.com/elastic/kibana/assets/92328789/fdfb6761-cc08-46dc-8032-9babd15e40c1">



https://github.com/elastic/kibana/blob/a4f8e9e7f10d632e74f6f87258d21f579c4e2bec/x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/select_rule_type/index.tsx#L197

I think this change should do it

```ts
          {((!isUpdateView && isEsqlRuleTypeEnabled) || esqlSelectableConfig.isSelected) && (
```